lib/ansible/plugins/action/net_base.py

```diff
@@ -73,8 +78,23 @@ def run(self, tmp=None, task_vars=None):
                 play_context.become_method = 'enable'
 
             if self._play_context.connection == 'local':
-                socket_path = self._start_connection(play_context)
-                task_vars['ansible_socket'] = socket_path
+                if self.provider['transport'] == 'nxapi' and play_context.network_os == 'nxos':
+                    obj = {}
+                    obj['remote_addr'] = self._play_context.remote_addr
+                    obj['connection_user'] = self._play_context.connection_user
+                    obj['password'] = self._play_context.password
```

Comment:
Can setting above three keys moved in  `nxapi_implementation()`

Reply:
Same like above.